### PR TITLE
New features

### PR DESCRIPTION
**Pull request by JonathanPaetzold**

### CODE REVIEW

File: Assets/Scripts/Abilities.cs [hunk 1]
```diff
@@ -214,4 +214,328 @@ private IEnumerator _flashlightAbility(PlayerCharacterStats stats)
             EnableButtons();
         }
     }
+
+
+	public void CoveringFire(PlayerCharacterStats stats) { if (stats.Actionsleft > 0) StartCoroutine(_CoveringFire(stats)); }
+	private IEnumerator _CoveringFire(PlayerCharacterStats stats)
```
Comment (vogel-sa) [inline Assets/Scripts/Abilities.cs:254]: This doesn't appear to do anything.

File: Assets/Scripts/Abilities.cs [hunk 2]
```diff
@@ -214,4 +214,328 @@ private IEnumerator _flashlightAbility(PlayerCharacterStats stats)
...
+			do
+			{
+				if (Input.GetKeyDown(KeyCode.Escape)) yield break;
+				yield return null;
+			} while (!Input.GetMouseButtonDown(0));
+			stats.isCoveringFire = true;
+			abilData.Currcooldown = abilData.Maxcooldown;
+			//stats.hasMoved = true;
+			stats.Actionsleft--;
+			GetComponent<TurnManager>().AutoEndTurnCheck();
```
Comment (vogel-sa) [inline Assets/Scripts/Abilities.cs:280]: Why is this check being made?

File: Assets/Scripts/Abilities.cs [hunk 3]
```diff
@@ -214,4 +214,328 @@ private IEnumerator _flashlightAbility(PlayerCharacterStats stats)
...
+			//Debug.Log(abilData.Description);
+			GetComponent<PlayerMovementManager>().SetQuadsEnabled(false);
+			GetComponent<PlayerMovementManager>().enabled = false;
+			do
+			{
+				if (Input.GetKeyDown(KeyCode.Escape)) yield break;
+				yield return null;
+			} while (!Input.GetMouseButtonDown(0));
+
+			stats.MitigationValue = 2;
```
Comment (vogel-sa) [inline Assets/Scripts/Abilities.cs]: This will never wear off.
Also, why are you just setting it to 2?

**Review by vogel-sa — CHANGES_REQUESTED** [review]
I've made several comments, please address them.

File: Assets/Scripts/Abilities.cs (hunk 1 above)
Comment (JonathanPaetzold) [reply Assets/Scripts/Abilities.cs:254]: The ability is setting the player to be on covering fire and the effects are in enemy turn script, where the enemies check if they are in range and line of sight of a player who is using covering fire and take damage if they are

File: Assets/Scripts/Abilities.cs (hunk 3 above)
Comment (JonathanPaetzold) [reply Assets/Scripts/Abilities.cs]: I used the 2 as a quick test when making it, I thought the change to making based of ability data was in my push.  I'll make sure it is changed and fix it to wear off

File: Assets/Scripts/Abilities.cs (hunk 2 above)
Comment (JonathanPaetzold) [reply Assets/Scripts/Abilities.cs:280]: This is the same check as the other abilities making sure players don't move while trying to confirm the ability and to auto check for the end of the turn.  Not shown here but this also now does the UI for the ability.

**Review by vogel-sa — APPROVED** [review]

**Pull request merged**